sale_loyalty_partner_applicability/models/sale_order.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import ast
+
+from odoo import _, models
+from odoo.osv import expression
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_partner_domain(self, rule, partner_id):
+        # For now, let's return the rule's partner domain.
+        if rule.rule_partners_domain != "[]":
+            return expression.AND(
+                [
+                    ast.literal_eval(rule.rule_partners_domain),
+                    [("id", "=", partner_id.id)],
```

Comment:
I'm think that we could get rid of the modules `*_coupon_commercial_partner_applicability` adding a simple setting here (or in the base one being purists) to check just the partner or the commercial partner children. With that we would save three modules including the glue `coupon_commercial_partner_applicability`.

Reply:
I have added the functionality of the coupon_commercial_partner_applicability module by adding an option in settings in the sales section. In addition, I have extended the module description and usage as well as added tests to check this added functionality.

Reply:
As a clarification it is necessary to make the check in _try_apply_code for when a code is applied, because if it is done later the code will already be applied and even if there is an error it will be at promotion level and not at coupon level.